Unable to build application

I am trying to build my application after adding your threads module when I build I am getting the following error.

>:react-native-threads:incrementalReleaseJavaCompilationSafeguard UP-TO-DATE
:react-native-threads:compileReleaseJavaWithJavac
:react-native-threads:compileReleaseJavaWithJavac - is not incremental (e.g. outputs have changed, no previous execution, etc.).
/Users/sjain/Documents/Native_App/node_modules/react-native-threads/android/src/main/java/com/reactlibrary/ThreadBaseReactPackage.java:38: error: constructor AndroidInfoModule in class AndroidInfoModule cannot be applied to given types;
                new AndroidInfoModule(),
                ^
  required: ReactApplicationContext
  found: no arguments
  reason: actual and formal argument lists differ in length
1 error
:react-native-threads:compileReleaseJavaWithJavac FAILED
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':react-native-threads:compileReleaseJavaWithJavac'.


I tried to do react-native link but thing is I am also using `react-native-navigation` which has bit different implementation of `MainApplication.java` file and thus its unable to get the context of ReactApplication.

Below is the implementation of my MainApplication.java file
`
public class MainApplication extends NavigationApplication {

    @Override
    public boolean isDebug() {
        // Make sure you are using BuildConfig from your own application
        return BuildConfig.DEBUG;
    }

    protected List<ReactPackage> getPackages() {
        // Add additional packages you require here
        // No need to add RnnPackage and MainReactPackage
        return Arrays.<ReactPackage>asList(
            new RNThreadPackage()
        );
    }

    @Override
    public List<ReactPackage> createAdditionalReactPackages() {
        return getPackages();
    }

    @Override
    public String getJSMainModuleName() {
        return "index";
    }
   }
`

--- a/android/src/main/java/com/reactlibrary/ThreadBaseReactPackage.java
+++ b/android/src/main/java/com/reactlibrary/ThreadBaseReactPackage.java
@@ -35,7 +35,7 @@ public class ThreadBaseReactPackage implements ReactPackage {
     public List<NativeModule> createNativeModules(ReactApplicationContext catalystApplicationContext) {
         return Arrays.<NativeModule>asList(
                 // Core list
-                new AndroidInfoModule(),
+                new AndroidInfoModule(catalystApplicationContext),
                 new ExceptionsManagerModule(reactInstanceManager.getDevSupportManager()),
                 new AppStateModule(catalystApplicationContext),
                 new Timing(catalystApplicationContext, reactInstanceManager.getDevSupportManager()),

